superset-frontend/src/explore/actions/saveModalActions.js

```diff
@@ -76,18 +81,24 @@ export const getSlicePayload = (
   // would end up as an extra filter and when overwriting the chart the original
   // time range adhoc_filter was lost
   if (isEmpty(adhocFilters?.adhoc_filters) && !isEmpty(formDataFromSlice)) {
-    adhocFilters = extractAddHocFiltersFromFormData(formDataFromSlice);
+    formDataFromSlice?.adhoc_filters?.forEach(filter => {
```

Comment:
@lilykuang the comment states:

> Retain adhoc_filters from the slice if no adhoc_filters are present

but now we're only retaining temporal filters where `!isExtra`. Is this correct or do we also need to retain adhoc filters that are not temporal? If this is correct, can you adjust the comment to reflect the change?